extensions/interactions/MultipleChoiceInput/directives/oppia-interactive-multiple-choice-input.component.ts

```diff
@@ -114,6 +161,14 @@ export class InteractiveMultipleChoiceInputComponent implements OnInit {
     // Selected current option.
     (event.currentTarget as HTMLDivElement).classList.add('selected');
     this.answer = parseInt(answer, 10);
+    // Store current choice order in question element to
+    // prevent shuffling of choices after the first selection attempt.
+    let questionElement = document.querySelector(
+      '.oppia-rte-viewer.oppia-learner-view-card-top-content');
+    if (questionElement) {
+      questionElement.setAttribute(
+        'oppia-mcq-choice-order', JSON.stringify(this.orderOfChoices));
```

Comment:
@paprajapati9 when do we unset this? We should avoid directly manipulating the DOM because Angular does not track it, it can result in unexpected issues. These changes seem to have resulted in a bug that is causing the MCQ interaction to show shuffled choices even though the customisation arg for it specifies that it shouldn't shuffle. Here's a video showing the problematic behaviour:


https://user-images.githubusercontent.com/11008603/186263327-77739de3-2952-426a-89d7-dfaaff224824.mp4



The first occurence of the MCQ interaction is expected to shuffle choices based on the customisation args and the second is not supposed to. The attribute set for the first interaction persists for the second occurence and orders the choices based on that.

Reply:
I understand that DOM manipulation should not be done, but I don't think based on this logic the shuffling should happen. Because the should only happen if `showChoicesInShuffledOrder.value` is true based on the logic below.
`this.choices = (
        showChoicesInShuffledOrder.value ? shuffleChoices(choicesWithIndex) :
        choicesWithIndex.sort((c1, c2) => c1.originalIndex - c2.originalIndex)
      );`

Reply:
The value of "oppia-mcq-choice-order" persists so when there are multiple MCQ questions in the exp, it may refer to incorrect data.

Reply:
The value for `questionIsAnsweredOnce` is true after the first time a MCQ question is answered. So it executes line 99 - 106 for subsequent attempts and uses the order set in  "oppia-mcq-choice-order" attribute -- which may be stale.